### PR TITLE
Dynamic courses page

### PR DESCRIPTION
**Pull request by Joephy527**
all of the static files used are going to be replaced so i haven't followed naming guidelines for images.

parts of the ui like reading materials and assessment quizes haven't been added

here is the rest of the ui for different screen sizes and in different states

mobile
mobile-side
tablet
tablet-side
screencapture-localhost-3000-courses-asda-1-2024-02-07-07_28_38
screencapture-localhost-3000-courses-asda-1-2024-02-07-07_28_09

### CODE REVIEW

File: app/courses/[courseId]/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}: {
+  children: React.ReactNode;
+  params: { courseId: string; contentId: string };
+}) {
+  return (
+    <html lang="en">
+      <body>
+        <main className="scroll">
+          <Hero />
+
```
Comment (BrookW-83) [inline app/courses/[courseId]/layout.tsx:25]: - Add comments for this section in the white space

File: app/courses/[courseId]/layout.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          <section className="md:px-10 px-5">
+            <Heading />
+          </section>
+
+          <section className="xl:flex">
+            <SideBar id={params.courseId} />
+
+            {children}
```
Comment (BrookW-83) [inline app/courses/[courseId]/layout.tsx:37]: - Add comment here as well

File: app/components/dynamic-course/SideBar.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+              key={data.key}
+              src={data.src}
+              vidLength={data.vidLength}
+              title={data.title}
+              part={data.part}
+              status={data.status}
+            />
+          </Link>
+        ))}
+
```
Comment (BrookW-83) [inline app/components/dynamic-course/SideBar.tsx:77]: - remove this white space
-

**Review by BrookW-83 — CHANGES_REQUESTED** [review]
- The work is clean, and has few errors
- The bracnh says it has merge conflict , so try to resolve that

File: app/components/dynamic-course/SideBar.tsx (hunk 3 above)
Comment (Joephy527) [reply app/components/dynamic-course/SideBar.tsx:77]: intentional line break between sibling elements for better code readability

**Comment by Joephy527** [comment]
fixed merge conflict, just waiting approval

**Review by BrookW-83 — CHANGES_REQUESTED** [review]
- The branch has conflicts try to resolve them

**Review by BrookW-83 — APPROVED** [review]

**Pull request merged**